### PR TITLE
CLI Testing

### LINKED ISSUE #135: Test the CLI
We should test the unblob CLI.

We could do by simply executing the `unblob` command ... 
.... but we could also check this out, since we are using `click` for everything CLI-related: https://click.palletsprojects.com/en/8.0.x/testing/

### PR DESCRIPTION
**Pull request by kukovecz**
Fixes #135

### CODE REVIEW

File: tests/test_cli.py [hunk 1]
```diff
@@ -39,3 +43,110 @@ def test_show_external_dependencies_not_exists(monkeypatch):
...
+            ["--verbose", "--extract-dir", "unblob", "--depth", "2", "tests", "--help"],
+            id="eager_2",
+        ),
+    ),
+)
+def test_help(params):
+    runner = CliRunner()
+    result = runner.invoke(unblob.cli.cli, params)
+    assert result.exit_code == 0
+    # NOTE: In practive, it writes "Usage: unblob ...", this is done in the `cli.main` with `click.make_context`
```
Comment (QKaiser) [inline tests/test_cli.py:66]:
```suggestion
    # NOTE: In practice, it writes "Usage: unblob ...", this is done in the `cli.main` with `click.make_context`
```

**Review by QKaiser — APPROVED** [review]
Looks good to me.

File: tests/test_cli.py [hunk 2]
```diff
@@ -39,3 +43,110 @@ def test_show_external_dependencies_not_exists(monkeypatch):
...
+        pytest.param(
+            ["--verbose", "--extract-dir", "unblob", "--depth", "2"],
+            id="verbose+extract-dir+depth",
+        ),
+    ),
+)
+def test_without_file(params: List[str]):
+    runner = CliRunner()
+    result = runner.invoke(unblob.cli.cli, params)
+    assert result.exit_code == 0
```
Comment (vlaci) [inline tests/test_cli.py:88]: I think we'd want a non-zero exit code in that case

File: unblob/cli.py [hunk 3]
```diff
@@ -74,7 +74,11 @@ def get_help_text():
 )
 def cli(files: Tuple[Path], extract_root: Path, depth: int, verbose: bool):
     configure_logger(verbose, extract_root)
-    logger.info("Start processing files", count=noformat(len(files)))
+    file_count = len(files)
+    if file_count <= 0:
```
Comment (vlaci) [inline unblob/cli.py:78]: I'd just set `required=True` for the files argument

File: tests/test_cli.py [hunk 4]
```diff
@@ -39,3 +43,110 @@ def test_show_external_dependencies_not_exists(monkeypatch):
...
+    in_path = (
+        Path(__file__).parent
+        / "integration"
+        / "archive"
+        / "zip"
+        / "regular"
+        / "__input__/"
+    )
+    new_params = params + ["--extract-dir", str(tmp_path), str(in_path)]
+    result = runner.invoke(unblob.cli.cli, new_params)
```
Comment (vlaci) [inline tests/test_cli.py:124]: I'd mock the `process_file` call and assert on its arguments. Wouldn't run this with any of the real integration test assets.

**Review by vlaci — CHANGES_REQUESTED** [review]

File: unblob/cli.py (hunk 3 above)
Comment (kukovecz) [reply unblob/cli.py:78]: Yeah, good idea!

**Review by vlaci — APPROVED** [review]

**Pull request merged**